apps/src/weblab2/CDOIDE/types.ts

```diff
@@ -0,0 +1,69 @@
+export type LeftNavElement = {
+  icon: string;
+  component: string;
+};
+
+export type FileId = string;
+export type FolderId = string;
+
+export type PreviewComponent = (args: {file: ProjectFileType}) => JSX.Element;
+export type EditorComponent = () => JSX.Element;
+export type EmptyEditorComponent = () => JSX.Element;
+
+export type SideBarItem = {
+  icon: string;
+  action: () => void;
+  label?: string;
+};
+
+export type ConfigType = {
+  showLeftNav?: boolean;
+  showEditor?: boolean;
+  showPreview?: boolean;
+  showSideBar?: boolean;
+  showRunBar?: boolean;
+  showDebug?: boolean;
+  activeLeftNav: string;
+  sideBar: SideBarItem[];
+  instructions: string | undefined;
+  defaultTheme?: EditorTheme;
+  leftNav: LeftNavElement[];
+  EditorComponent?: EditorComponent;
+  editableFileTypes?: string[];
+  previewFileTypes?: string[];
+  EmptyEditorComponent?: EmptyEditorComponent;
+  blankEmptyEditor?: boolean;
+  PreviewComponents?: {[key: string]: PreviewComponent};
+};
+
+export type ProjectFolderType = {
+  id: FolderId;
+  name: string;
+  parentId: string;
+  open?: boolean;
+};
+
+export type ProjectFileType = {
+  id: FileId;
+  name: string;
+  language: string;
+  contents: string;
+  open?: boolean;
+  active?: boolean;
+  folderId: string;
+};
+
+export type ProjectType = {
```

Comment:
let's use the lab2/types version of this instead (called `MultiFileSource`)

Reply:
Done!